src/debug.cpp

```diff
@@ -140,53 +141,42 @@ void DebugPrint(const char *category, int level, std::string &&message)
  * @param s Text describing the wanted debugging levels.
  * @param error_func The function to call if a parse error occurs.
  */
-void SetDebugString(const char *s, void (*error_func)(const std::string &))
+void SetDebugString(std::string_view s, SetDebugStringErrorFunc error_func)
 {
-	int v;
-	char *end;
-	const char *t;
+	StringConsumer consumer{s};
 
 	/* Store planned changes into map during parse */
-	std::map<const char *, int> new_levels;
+	std::map<std::string_view, int> new_levels;
 
 	/* Global debugging level? */
-	if (*s >= '0' && *s <= '9') {
-		v = std::strtoul(s, &end, 0);
-		s = end;
-
+	auto level = consumer.TryReadIntegerBase<int>(10);
+	if (level.has_value()) {
 		for (const auto &debug_level : _debug_levels) {
-			new_levels[debug_level.name] = v;
+			new_levels[debug_level.name] = *level;
 		}
 	}
 
 	/* Individual levels */
-	for (;;) {
+	while (consumer.AnyBytesLeft()) {
 		/* skip delimiters */
-		while (*s == ' ' || *s == ',' || *s == '\t') s++;
-		if (*s == '\0') break;
+		consumer.SkipUntilCharNotIn(" \t,");
 
-		t = s;
-		while (*s >= 'a' && *s <= 'z') s++;
+		std::string_view key = consumer.ReadUntilChar('=', StringConsumer::SKIP_ONE_SEPARATOR);
```

Comment:
The old code also stopped at e.g. whitespace.
Now whitespace is included in `key`, so it does not match any debug level.